### ISSUE TITLE
No ability to set minimum diff. for individual miner (p2pool jobs have too high diff. for slow miners)

### ISSUE THREAD

**Issue by xmrvsbeast** [SChernykh/p2pool#4]
It would be useful in some situations to be able to set a minimum difficulty on an individual miner, for example passing a "d=500000" as a password parameter when connecting to p2pool stratum server

Some miners will not start hashing if min diff is too low but other miners will not work if min diff is too high.

Support for lower min diff than 100,000 would also be nice to have, ideally all the way down to 10,000 or even 1000 if possible.

There appears to be an issue with p2pool setting a proper diff.  When I try mining to p2pool with a slow miner ~650h/s It is sending jobs with 309140 diff when it should be about 10 times less.  

Is the diff of the p2pool share chain not as flexible on diff as compared to what a traditional pool can do?

thanks

**Comment by SChernykh** [comment]
Added support in latest:

`-u x+10000` in the command line, or
```
    "pools": [
        {
            "url": "127.0.0.1:3333",
            "user": "x+10000",
            "enabled": true,
        }
    ],
```
in config.json

**Comment by xmrvsbeast** [comment]
This is great, however I am getting an error when putting 10000 in the config.json. 

SideChain min_diff is invalid (must be between 100000 and 1000000000)

It would be beneficial to be able to set a min start diff as well not just fixed diff

For example p2pool is set to 10000 min dynamic start diff

One miner is set to fd+50000 fixed diff

another miner is set to dd+1000000 min start diff but still dynamic

thank you

**Comment by SChernykh** [comment]
> SideChain min_diff is invalid (must be between 100000 and 1000000000)

This is not the diff for miners, this is sidechain consensus settings. If you change it, you'll be mining on a different p2pool.

**Comment by xmrvsbeast** [comment]
> > SideChain min_diff is invalid (must be between 100000 and 1000000000)
> 
> This is not the diff for miners, this is sidechain consensus settings. If you change it, you'll be mining on a different p2pool.

OK, thank you for the explanation

**Comment by SChernykh** [comment]
@xmrvsbeast I've restarted p2pool with new sidechain. Don't worry, your hashrate will keep mining on the old p2pool and get Monero blocks until you switch - I literally can't stop it because it's p2pool :) Can you build the latest code and follow new instructions (they're the same, but port number changed to 37889)? I've fixed a critical bug with the code that generates miner reward transaction, but it required to start a new sidechain.

**Comment by xmrvsbeast** [comment]
Thanks for the notice, is there a way to unblock peers?

P2PServer peer 65.21.227.114:37889 is banned, not connecting to it

**Comment by SChernykh** [comment]
Your node was banned because of incompatible config.json. But bans are temporary (10 minutes), so you were able to connect eventually.

> Maybe it would be good to include the latest p2pool_peers.txt in the repo

I will do it for the release.

**Comment by SChernykh** [comment]
@xmrvsbeast can you update your node to the latest code again? Also, what's the best way to contact you? You're not on IRC...
